### PR TITLE
first pass at admin docs for self registration service

### PR DESCRIPTION
**Pull request by jscearce3**
## Description

Adding more clarity for administrators of a Nebari deployment on how to install and configure the plugin.

## Related Issue

https://gitlab.jatic.net/jatic/team-metrostar/t-e-platform/-/issues/544

## Motivation and Context

Better documentation around plugin usage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,13 +8,38 @@
...
+In order to install this plugin as part of a Nebari deployment:
+- Create a conda environment for your Nebari deployment
+- Install the self registration plugin with `pip install nebari-plugin-self-registration`
+- Continue the initialization and deployment of Nebari per your provider [instructions](https://www.nebari.dev/docs/explanations/provider-configuration).
+
+> **NOTE:** When running `nebari render` and `nebari deploy`, Nebari will detect and install any extensions which are installed in your Python environment.  When managing multiple Nebari deployments, be sure to manage your conda environments to ensure the correct extensions and versions are installed in your target deployment.
+
+
+### Configuration
+The configuration of your self registration app can be customized in several ways by creating a `config.yaml` file within the `self-registration` folder. An example of a configuration yaml is provided [here](https://github.com/MetroStar/nebari-self-registration/blob/main/self-registration/config.sample.yaml).
```
Comment (marcelovilla) [inline README.md:26]: Where should be the `self-registration` folder? Same level as `nebari-config.yaml`?

When I installed the plugin, I added a section like this to the `nebari-config.yaml` file:
```yaml
self_registration:
  namespace: self-registration
  coupons:
    - abcdefg
  approved_domains:
    - metrostar.com
  account_expiration_days: 30
  registration_group: test-group
```

Would it be worth to add this as another option in the docs for users that prefer to keep all the configuration within a single file?

File: README.md [hunk 2]
```diff
@@ -8,13 +8,38 @@
...
+
+Configuration options include:
+
+
+- **account_expiration_days (optional)**: Days a user is provided to verify their account.  Defaults to 7.
```
Comment (marcelovilla) [inline README.md:31]: I think this is rather the number of days after which the user is disabled in keycloak.

File: README.md [hunk 3]
```diff
@@ -8,13 +8,38 @@
...
+- **approved_domains (required)**: List of approved email domains that can register accounts using the self registration service.  (supports names like `gmail.com` and wildcards such as `*.edu`)
```
Comment (marcelovilla) [inline README.md:32]: Although it might be obvious, I'd add an example of using `*` to allow any domain.

**Review by marcelovilla — COMMENTED** [review]
Now that https://github.com/MetroStar/nebari-self-registration/pull/12 has been merged, would it make sense to include a brief description on how to configure custom themes?

File: README.md (hunk 3 above)
Comment (kenafoster) [reply README.md:32]: Fair enough - added!

**Comment by kenafoster** [comment]
Really good catches and suggestions @marcelovilla - you mind checking one more time with my latest commit?

**Review by marcelovilla — APPROVED** [review]
LGTM! Thanks for these additions!

**Pull request merged**